### ISSUE TITLE
Add support for --preserve-fds to podman run

### ISSUE THREAD

**Issue by eriksjolund** [containers/libpod#6458]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**
__Summary__: Add support for `--preserve-fds=N` to `podman run`, in the same way that the command-line option is supported in `podman exec`.

__Example__:
The command __foobar__ (in the container _localhost/example_) reads input from the file descriptor 0 and the file descriptor 3.
My goal is is to run __foobar__ and pass the file _file1.txt_ as file descriptor 0 and the file _file2.txt_ as file descriptor 3. (Both _file1.txt_ and _file2.txt_ are located on the host system)

It would be nice to be able to run

``` 
$ cat file1.txt | podman run --rm -i --preserve-fds=1 localhost/example foobar 3< file2.txt
```
but that is currently not possible. Instead I need  to use `podman exec`, something like this:

``` 
$ ctr=$(podman run -d --rm localhost/example sleep 300)
$ cat file1.txt | podman exec -i --preserve-fds=1 $ctr foobar 3< file2.txt
```

More info:

```
$ echo $0
-bash
$ podman --version
podman version 1.9.2
```

**Comment by mheon** [comment]
Shouldn't be that difficult (the reason `--preserve-fds` is exec-only is that the original request was only for exec, there are no technical reasons it won't work with `podman run`). Unfortunately, most of the team is swamped with Podman 2.0 work at the moment, so I can't guarantee we'll get to this quickly; community PRs are always welcome, though!

**Comment by rhatdan** [comment]
@eriksjolund Feel like opening a PR?
@sujil02 @QiWang19 PTAL

**Comment by eriksjolund** [comment]
@rhatdan Up till now I've only been a Golang code reader and not a Golang code writer, so it would take a bit of an effort to do an PR. Unfortunately I have too little time right now.

**Comment by rhatdan** [comment]
No problem @eriksjolund, thanks for the issue.

I am always trying to encourage contributions.  Anyone helping in OpenSource projects are appreciated.

**Comment by QiWang19** [comment]
@eriksjolund can you provide more details? what the uses case `foobar` command specifically running?

**Comment by eriksjolund** [comment]
> @eriksjolund can you provide more details? what the uses case `foobar` command specifically running?

I wanted to create a command-line tool to demonstrate a C++ library. The command-line tool needs two input files.
I think a user without much experience of bind mounts and SELINUX security file contexts would find it convenient to be able to do:

```
cat file1.txt | podman run --rm -i --preserve-fds=1 localhost/example foobar 3< file2.txt
```


I see that the PR https://github.com/containers/podman/pull/6625 has been merged:
Thanks!

Is the feature available in any release?

### Fedora 32

```
[erik@laptop ~]$ cat /etc/fedora-release 
Fedora release 32 (Thirty Two)
[erik@laptop ~]$ podman --version
podman version 2.0.2
[erik@laptop ~]$ podman run --help | grep preserve-fds
[erik@laptop ~]$ 
```
### Centos 8

```
[erik@centos ~]$ cat /etc/centos-release
CentOS Linux release 8.2.2004 (Core) 
[erik@centos ~]$ podman --version
podman version 2.0.3
[erik@centos ~]$ podman run --help | grep preserve-fds
[erik@centos ~]$ rpm -qi podman
Name        : podman
Epoch       : 0
Version     : 2.0.3
Release     : 1.el8
Architecture: x86_64
Install Date: Thu 23 Jul 2020 11:56:28 PM CEST
Group       : Unspecified
Size        : 74267642
License     : ASL 2.0
Signature   : RSA/SHA256, Thu 23 Jul 2020 09:42:24 PM CEST, Key ID 4d64390375060aa4
Source RPM  : podman-2.0.3-1.el8.src.rpm
Build Date  : Thu 23 Jul 2020 09:41:48 PM CEST
Build Host  : cloud107
Relocations : (not relocatable)
Vendor      : obs://build.opensuse.org/devel:kubic
URL         : https://podman.io/
Summary     : Manage Pods, Containers and Container Images
Description :
podman (Pod Manager) is a fully featured container engine that is a simple
daemonless tool.  podman provides a Docker-CLI comparable command line that
eases the transition from other container engines and allows the management of
pods, containers and images.  Simply put: alias docker=podman.
Most podman commands can be run as a regular user, without requiring
additional privileges.

podman uses Buildah(1) internally to create container images.
Both tools share image (not container) storage, hence each can use or
manipulate images (but not containers) created by the other.

Manage Pods, Containers and Container Images
podman is a simple management tool for pods, containers and images
[erik@centos ~]$ 
```

**Comment by eriksjolund** [comment]
I found the answer myself.
`podman run --preserve-fds` might become available in v2.1.0 according to
https://github.com/containers/podman/pull/6909#issuecomment-663111712
